### PR TITLE
add sessionID and timestamp to analytics

### PR DESCRIPTION
**Pull request by wdwatkins**
I haven't implemented this for all events yet, so don't merge.  I wanted to get a review first.

I've added the `getTimestamp` and `getSessionId` functions.  `getTimestamp` is a hit-level dimension, so will record the time of every event.  `getSessionId` just needs to be a unique random number that represents the session.  As implemented now, it will be different for each hit, but because it is scoped as a session dimension, Google will only store the last value it receives and apply that to the entire session.  I wanted originally to just make `sessionId` a fixed variable, since it doesn't need to change, but I was getting errors with it being undefined.  This is how it was done here: https://www.simoahava.com/analytics/improve-data-collection-with-four-custom-dimensions/#3-session-id so may be unavoidable.

### CODE REVIEW

File: js/map_utils.js [hunk 1]
```diff
@@ -98,10 +97,13 @@ function zoomToFromState(d, i, j, selection) {
   }
   
   // zoom to the new view
-  updateView(newView);
+  updateView(newView, fireAnalytics = true);
```
Comment (wdwatkins) [inline js/map_utils.js]: analytics weren't firing at all on `updateView()`

File: js/map_utils.js [hunk 2]
```diff
@@ -98,10 +97,13 @@ function zoomToFromState(d, i, j, selection) {
...
 }
 
-function updateView(newView, fireAnalytics) {
+function getSessionId(){return new Date().getTime() + '.' + Math.random().toString(36).substring(5)}
```
Comment (aappling-usgs) [inline js/map_utils.js]: session ID is a timestamp?

Comment (wdwatkins) [reply js/map_utils.js]: With some appended random characters.  It's just a way of ensuring a random number.

Comment (wdwatkins) [reply js/map_utils.js]: To be more specific, it is a timestamp in milliseconds since 1970-01-01

Comment (aappling-usgs) [reply js/map_utils.js]: Cool, thanks :+1:

**Pull request merged**